api/python/arrus/utils/tests/imaging/preprocessing_test.py

```diff
@@ -0,0 +1,639 @@
+            kwargs["data"] = data
+        result = super().run_op(**kwargs)
+        return np.squeeze(result)
+
+    def test_clip(self):
+        # Given
+        data = np.arange(5)
+        # Run
+        result = self.run_op(data=data,min=2,max=3)
+        # Expect
+        expected = np.array([2, 2, 2, 3, 3])
+        np.testing.assert_equal(result, expected)
+
+
+class ToGrayscaleImgTestCase(ArrusImagingTestCase):
+
+    def setUp(self) -> None:
+        self.op = ToGrayscaleImg
+        self.context = self.get_default_context()
+
+    def run_op(self, **kwargs):
+        data = kwargs['data']
+        data = np.array(data)
+        if len(data.shape) > 3:
+            raise ValueError("Currently data supports at most 3 dimensions.")
+        if len(data.shape) < 3:
+            dim_diff = 3-len(data.shape)
+            data = np.expand_dims(data, axis=tuple(np.arange(dim_diff)))
+            kwargs["data"] = data
+        result = super().run_op(**kwargs)
+        return np.squeeze(result)
+
+    def test_is_in_range(self):
+        # Given
+        data = np.arange(-10, 1024)
+        # Run
+        result = self.run_op(data=data)
+        # Expect
+        self.assertTrue((np.all(result >= 0))
+                        and (np.all(result <= 255))
+                        and result.dtype == 'uint8')
+
+    def test_is_correct(self):
+        # Given
+        data = np.arange(-128, 128)
+        # Run
+        result = self.run_op(data=data)
+        # Expect
+        expected = data - np.min(data)
+        expected = expected/np.max(expected)*255
+        np.testing.assert_equal(expected, result)
+
+
+class ScanConversionLinearArrayTestCase(ArrusImagingTestCase):
+
+    def setUp(self) -> None:
+        sequence = LinSequence(
+            tx_aperture_center_element=np.arange(0, 65, 8),
+            tx_aperture_size=32,
+            rx_aperture_center_element=np.arange(0, 65, 8),
+            rx_aperture_size=32,
+            tx_focus=50e-6,
+            pulse=Pulse(center_frequency=6e6, n_periods=2,
+                        inverse=False),
+            rx_sample_range=(0, 2048),
+            downsampling_factor=1,
+            speed_of_sound=1490,
+            pri=100e-6,
+            sri=50e-3,
+            tgc_start=0,
+            tgc_slope=12,
+        )
+        self.op = ScanConversion
+        self.context = self.get_default_context(sequence=sequence)
+
+    def run_op(self, **kwargs):
+        data = kwargs['data']
+        data = np.array(data)
+        if len(data.shape) > 2:
+            raise ValueError("Currently data supports at most 2 dimensions.")
+        if len(data.shape) < 2:
+            dim_diff = 2-len(data.shape)
+            data = np.expand_dims(data, axis=tuple(np.arange(dim_diff)))
+            kwargs["data"] = data
+        result = super().run_op(**kwargs)
+        return np.squeeze(result)
+
+    def test_identity(self):
+        # Given
+        n_scanlines = get_n_scanlines(self)
```

Comment:
It would be better if this function was a class method (instead of a package function which accesses class fields), i.e. it's preferred to use `self.get_n_scanlines()` instead of `get_n_scanlines(self)`

1. Create a common base class for all ScanConversionTests, e.g. `AbstractScanConversionTestCase`
2. Move the `run_op`, get_n_scanlines`, etc. to the common base class.
3. Inherit the base class for ScanConversionLinearArrayTestCase and ScanConversionConvexTestCase, use the base class methods.

Reply:
It is done (I suppose...).